cetmix_tower_server/tests/test_plan.py

```diff
@@ -1261,3 +1261,503 @@ def test_plan_with_another_plan_with_all_not_executable_condition(self):
+        self.remove_from_group(
+            self.user_bob,
+            [
+                "cetmix_tower_server.group_user",
+                "cetmix_tower_server.group_manager",
+                "cetmix_tower_server.group_root",
+            ],
+        )
+        self.write_and_invalidate(self.command_create_dir, **{"access_level": "1"})
+
+        # Ensure that user_bob without any group cannot access plan line actions
+        test_plan_line_action_as_bob = self.plan_line_action.with_user(self.user_bob)
+        with self.assertRaises(AccessError):
+            plan_line_action_read_result = test_plan_line_action_as_bob.read([])
+
+        # Add user_bob to `group_user` and test plan.line.action access
+        self.add_to_group(self.user_bob, "cetmix_tower_server.group_user")
+        # Set access level to 1, so group_user can access the plan line actions
+        self.write_and_invalidate(self.plan_2, **{"access_level": "1"})
+        self.assertEqual(self.plan_line_action.access_level, "1")
+
+        plan_line_action_read_result = test_plan_line_action_as_bob.read([])
+        self.assertEqual(
+            plan_line_action_read_result[0]["condition"],
+            test_plan_line_action_as_bob.condition,
+            msg="User should access plan line actions with access_level 1",
+        )
+
+        # Add user_bob to `group_manager` and test plan.line.action access
+        self.add_to_group(self.user_bob, "cetmix_tower_server.group_manager")
+        # Set access level to 2, so group_manager can access the plan line action
+        self.write_and_invalidate(self.plan_2, **{"access_level": "2"})
+        # Ensure that user_bob as member of group_manager
+        # can read plan line actions
+        plan_line_action_read_result = test_plan_line_action_as_bob.read([])
+        self.assertEqual(
+            plan_line_action_read_result[0]["name"],
+            test_plan_line_action_as_bob.name,
+            msg="Name should be the same",
+        )
+
+        # Ensure that manager can update plan line actions they did not create
+        test_plan_line_action_as_bob.write({"sequence": 3})
+        self.assertEqual(
+            test_plan_line_action_as_bob.sequence,
+            3,
+            msg="Manager should be able to update sequence",
+        )
+
+        # Ensure that manager cannot delete plan line actions they did not create
+        with self.assertRaises(AccessError):
+            test_plan_line_action_as_bob.unlink()
+
+        # Create a new plan line action with user_bob as manager
+        new_plan_line_action = (
+            self.env["cx.tower.plan.line.action"]
+            .with_user(self.user_bob)
+            .create(
+                {
+                    "line_id": self.plan_2.line_ids[0].id,
+                    "condition": ">",
+                    "value_char": "100",
+                    "action": "e",
+                }
+            )
+        )
+
+        self.assertEqual(
```

Comment:
No need to check the ORM methods we don't modify.

Reply:
Fixed